mytoyota/models/vehicle.py

```diff
@@ -287,13 +346,40 @@ async def get_summary(
                     Summary(
                         summary.summary,
                         self._metric,
-                        summary_from_date if summary_from_date > from_date else from_date,
-                        summary_to_date if summary_to_date < to_date else to_date,
+                        # The data might not include an entire month so update start and end dates
+                        max(summary_from_date, from_date),
+                        min(summary_to_date, to_date),
                         summary.hdc,
                     )
                 )
         elif summary_type == SummaryType.YEARLY:
-            raise NotImplementedError
+            # TODO: Require confirmation that the endpoint can return data older than a year
```

Comment:
Hi @GitOldGrumpy, my car is from 2019. It seems that I can only retrieve data that goes back a maximum of 364 days. For today's date (30.12.2023), for example, the oldest trip for which I receive information is 31.12.2022.

Reply:
I expected as much but dont think it makes anything different to the code.